lib/msf/util/exe.rb

```diff
@@ -889,9 +895,10 @@ def self.to_osx_app(exe, opts = {})
     zip.add_file("#{app_name}/Contents/", '')
     zip.add_file("#{app_name}/Contents/MacOS/", '')
     zip.add_file("#{app_name}/Contents/Resources/", '')
-    zip.add_file("#{app_name}/Contents/MacOS/#{exe_name}", exe)
+    zip.add_file("#{app_name}/Contents/MacOS/#{exe_name}", File.new(macho, 'r').read)
```

Comment:
This will simply create a new "read" permissions file that is dumps the data into.  The file needs to be created with executable permissions or staged with the perms and added from disk in some way.

@thecarterb Try this in irb, shows that new file created at least on osx will be read not read & execute

```
require 'fileutils'
require 'rex/zip'

exe_name = "executable"
macho = File.new('test.local', 'w')
macho.write("testdata")
macho.close
FileUtils.chmod(0777, macho)

zip = Rex::Zip::Archive.new
zip.add_file("#{exe_name}", File.new(macho, 'r').read)
packed_data = zip.pack

packed = File.new('packed.zip', 'w')
packed.write(zip.pack)
packed.close
```

packed.zip does not end up with executable perms looking at Rex::Zip::Archive.add_file I am not sure this utility properly supports adding files with executable permission.  This looks like it may be possible with an ```attrs``` parameter however this does not look to be available as a current parameter to ```add_file```.